src/mongo/nodes.ts

```diff
@@ -185,9 +185,12 @@ export class MongoDatabaseNode implements INode {
 	}
 }
 
-export class MongoCollectionNode implements INode {
+export class MongoCollectionNode implements IEditableNode {
 
-	constructor(readonly collection: Collection, readonly db: MongoDatabaseNode) {
+	constructor(readonly collection: Collection, readonly db: MongoDatabaseNode, data: Array<any>, readonly query: string) {
+		data.forEach(element => {
```

Comment:
If you pass in data, then 'Load more...' won't work because the iterator isn't set. Instead, don't pass in data (don't even execute the command for `find`) and just call node.getChildren to fetch the first batch

(Down the line we might add a 'Load More...' right click menu to each file - in which case the iterator needs to be set properly)

Reply:
getChildren is async. I can't really call it from a property (get data()) or the constructor. 
For a find result, I don't see how we can call node.getChildren at the point of showDocument because we need to await it.